src/core/server/saved_objects/service/lib/utils.ts

```diff
@@ -69,4 +71,21 @@ export class SavedObjectsUtils {
     total: 0,
     saved_objects: [],
   });
+
+  /**
+   * Generates a random ID for a saved objects.
+   */
+  public static generateId() {
+    return uuid.v1();
+  }
```

Comment:
Perhaps we should use `uuid.v4()` here instead, considering that the ESO wrapper now relies on this?

I know that [you should not rely on UUIDs being hard to guess](https://tools.ietf.org/html/rfc4122#section-6), but UUIDv1 is very predictable where UUIDv4 is not. I don't see any practical benefit to using v1 over v4.

@pgayvallet do you have any thoughts on this either way?

Reply:
> Still, I think v1 is good enough, it 'just work (tm)', so there is no urge to change it (or do you see one?)

My only justification is that the ESO wrapper used to generate its own IDs upon object creation, and it used v4. Now it's relying on the core `generateId` function which uses v1.

I think it's worth discussing further. I'm curious to know why we're hesitant to change to v4 in a minor.